### PR TITLE
Implement Retries for elasticache discovery

### PR DESCRIPTION
**Pull request by michaeljguarino**
This really shouldn't fail, but when it does, we want to be more robust and not completely
nuke the cluster until another pool succeeds.

### CODE REVIEW

**Review by stephanos — APPROVED** [review]

File: lib/memcachir/util.ex [hunk 1]
```diff
@@ -3,6 +3,10 @@ defmodule Memcachir.Util do
   Utilities to read configuration.
   """
 
+  @retry_base 10
+  @jitter     10
+  @max_retries 3
```
Comment (stephanos) [inline lib/memcachir/util.ex]: maybe making these variables configurable would be a nice touch. or at least `max_retries`

Comment (michaeljguarino) [reply lib/memcachir/util.ex]: Sure that shouldn't be a problem.

**Comment by stephanos** [comment]
@peillis I'd be great if we could merge this :)

**Comment by peillis** [comment]
ok done

**Pull request merged**